optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/bavet/bi/BavetGroupBiTuple.java

```diff
@@ -29,6 +32,7 @@
     private int parentCount;
     private ResultContainer_ resultContainer;
     private Result_ result;
+    private final List<BavetAbstractTuple> childTupleList = new ArrayList<>();
```

Comment:
The bridge will have n childs typically, but the group/joins itself - like this one - will have 1 child typically.

```suggestion
    private final List<BavetAbstractTuple> childTupleList = new ArrayList<>(1);
```

Reply:
Hmm, I have it switched the other way. Will need rebenchmarking then, could be even better.